File: Python_Codes/15. Memory Management/FinalizeExample.py
import gc

class MyObject:
    def __init__(self, name):
        self.name = name

    def __del__(self):
        print(f"{self.name} is finalized and resources are released!")

def demonstrate_finalize():
    obj = MyObject("Finalizing Object")
    obj = None  # Eligible for garbage collection
    gc.collect()  # Suggest garbage collection

if __name__ == "__main__":
    demonstrate_finalize()

File: Projects/JavaProject/src/com/Memory_Management/FinalizeExample.java
package com.Memory_Management;

//Freeing resources like closing a file stream or releasing 
// database connections in server applications.

public class FinalizeExample {

    public static void demonstrateFinalize() {
        MyObject obj = new MyObject("Finalizing Object");
        //obj.finalize();
        obj = null; // Eligible for Garbage Collection
        
        System.gc(); // Suggest Garbage Collection
    }

    static class MyObject {
        private String name;

        public MyObject(String name) {
            this.name = name;
        }

        @Override
        protected void finalize() {
            System.out.println(name + " is finalized and resources are released!");
        }
    }

    public static void main(String[] args) {
        demonstrateFinalize();
    }
}

